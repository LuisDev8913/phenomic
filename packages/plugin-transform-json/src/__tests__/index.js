// @flow

import config from "@phenomic/core/lib/defaultConfig.js";

import transformJSON from "../index.js";

it("should transform json as json", () => {
  const plugin = transformJSON(config, {});
  expect(typeof plugin.transform === "function").toBe(true);
  expect(
    plugin.transform &&
      plugin.transform({
        file: {
          name: "file.json",
          fullpath: "/test/file.json",
          // exists: true,
          // type: "wat"
        },
        contents: new Buffer(
          `{
      "test": "a",
      "test2": "b",
      "somedata": "Normal version"
    }`,
        ),
      }),
  ).toMatchSnapshot();
});

it("should transform json as json and support partial", () => {
  const plugin = transformJSON(config, {});
  expect(typeof plugin.transform === "function").toBe(true);
  expect(
    plugin.transform &&
      plugin.transform({
        file: {
          name: "file.json",
          fullpath: "/test/file.json",
          // exists: true,
          // type: "wat"
        },
        contents: new Buffer(
          `{
      "test": "a",
      "test2": "b",
      "somedata": "Normal version",
      "partial": {
        "somedata": "Light version"
      }
    }`,
        ),
      }),
  ).toMatchSnapshot();
});
