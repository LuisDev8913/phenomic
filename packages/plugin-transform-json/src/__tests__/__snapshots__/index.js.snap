// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should transform json as json 1`] = `
Object {
  "data": Object {
    "somedata": "Normal version",
    "test": "a",
    "test2": "b",
  },
  "partial": Object {
    "somedata": "Normal version",
    "test": "a",
    "test2": "b",
    "title": "file.json",
  },
}
`;

exports[`should transform json as json and support partial 1`] = `
Object {
  "data": Object {
    "partial": Object {
      "somedata": "Light version",
    },
    "somedata": "Normal version",
    "test": "a",
    "test2": "b",
  },
  "partial": Object {
    "somedata": "Light version",
    "title": "file.json",
  },
}
`;
